eng/templates/stages/release.yml

```diff
@@ -148,6 +152,26 @@ stages:
             draft='--draft'
           fi
 
+          git_user=dotnet-bot
+          git_email=dotnet-bot@microsoft.com
+
+          git config --global user.name "$git_user"
+          git config --global user.email "$git_email"
+
+          git tag $releaseTag $dotnetDotnetCommit
+          
+          vmr_url=$(echo $vmrPublicUrl | sed "s,https://.*@,https://$(git_user):${GH_TOKEN}@,g")
+          result=$(git push $vmr_url refs/tags/$releaseTag 2>&1)
+          
+          if [[ $? != 0 ]]; then
+            echo "##vso[task.logissue type=error]Failed to create a GitHub tag for the release: $result"
+            exit 1
+          fi
+
+          if [[ "$result" =~ "Everything up-to-date" ]]; then
+            echo "##vso[task.logissue type=warning]GitHub release tag $(releaseTag) already exists ($(vmrPublicUrl)/releases/tag/$(releaseTag))"
```

Comment:
I would not warn about this. Instead, we could verify if the tag points to the same commit. If not, we should throw an error saying that the tag already exists and points to a different SHA.
Otherwise just log that we are good (this is a re-run and we already pushed the tag)

Reply:
I thought about this originally but could something like this happen? The tag in question points to a runtime version being released, so we would need to have two different commits of the VMR getting released with the same runtime version

Reply:
We can get additional commits for SB only that we would be released on top of what we release usually. This happened recently:
https://github.com/dotnet/installer/tree/release/8.0.1xx-preview3-sb

This is also when we then override the build IDs to configure what differs when we're releasing.

That being said, regardless of this flow, if it ever happens that we want to tag something and a tag already points to something else, we should definitely fail hard because this is unusual and we need to know about this.
So this check makes sense to me to have even if it looks like dead code.

Reply:
Sure thing, sounds reasonable, will add. Thanks for pointing it out

Reply:
Added the check